pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -257,6 +267,24 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 
 	ctx := context.Background()
 
+	// Try to retrieve the LogStorage resource to see if we need to clean up after it's been marked for deletion
+	if ls, err := GetLogStorage(ctx, r.client); err != nil {
+		if !errors.IsNotFound(err) {
+			r.status.SetDegraded("Failed to get LogStorage CR", err.Error())
+			return reconcile.Result{}, err
+		}
+	} else if ls.DeletionTimestamp != nil {
+		var errMsg string
+
+		err := r.finalizeDeletion(ctx, ls)
+		if err != nil {
+			errMsg = err.Error()
+		}
+
+		r.status.SetDegraded("Finalizing deletion of LogStorage before continuing", errMsg)
```

Comment:
Maybe I'm missing something, but won't this set degraded even if err == nil? Should we put this within the `err != nil` block above?

Reply:
I'm also a little bit confused about when `r.status.setDegraded` is meant to be used vs `r.setDegraded`

Reply:
You're not missing something, although I might not be doing the right thing. I thought that we would want to set the state as degraded since until we're finished cleaning up the resources.

I guess since it's there's nothing actually wrong maybe I shouldn't add this? I wanted to convey to the users that this is in a state of deletion in the case that it takes awhile for the ECK operator to clean up it's resources, but maybe logs are the way to go?

Reply:
> I'm also a little bit confused about when r.status.setDegraded is meant to be used vs r.setDegraded

I should be using r.setDegraded each time